flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -613,25 +614,31 @@ protected void cleanUpInvoke() throws Exception {
 		// stop all timers and threads
 		tryShutdownTimerService();
 
+		Throwable suppressedThrowable = null;
 		// stop all asynchronous checkpoint threads
 		try {
 			cancelables.close();
 			shutdownAsyncThreads();
 		} catch (Throwable t) {
-			// catch and log the exception to not replace the original exception
-			LOG.error("Could not shut down async checkpoint threads", t);
+			// catch and suppress the exception to not replace the original exception
+			suppressedThrowable = ExceptionUtils.firstOrSuppressed(t, suppressedThrowable);
 		}
 
 		// we must! perform this cleanup
 		try {
 			cleanup();
 		} catch (Throwable t) {
-			// catch and log the exception to not replace the original exception
-			LOG.error("Error during cleanup of stream task", t);
+			// catch and suppress the exception to not replace the original exception
+			suppressedThrowable = ExceptionUtils.firstOrSuppressed(t, suppressedThrowable);
 		}
 
 		// if the operators were not disposed before, do a hard dispose
-		disposeAllOperators(true);
+		try {
+			disposeAllOperators(true);
```

Comment:
I guess we can remove `logOnlyErrors` argument now as we are catching exceptions.

Reply:
I do not think so. Maybe we can change the argument name, but the meaning of logOnlyErrors true/false is different.

"False" means `disposalAllOperators` throws an exception as soon as it encounters an error.

"True" means `disposalAllOperators` disposes of all operators even though encountering errors, and then throw the stacked exceptions.

Reply:
You don't need "True" now because the call site catches the exceptions.

Reply:
As I said, the behavior is different.

I am not sure this should belong to this fix....

Reply:
The original issue is addressed by catching errors in `cleanUpInvoke`, right?

So the older change of suppressing in `disposalAllOperators` is no unnecessary. And then `logOnlyErrors` doesn't make sense.

If we keep the change then I don't see any reason why this should only happen for one branch (as I [mentioned above](https://github.com/apache/flink/pull/12525#discussion_r436595491)). Then `logOnlyErrors`  makes no sense again.

Reply:
remind: the same problem as the first question.